### ISSUE TITLE
Support reading empty frames with named index from tsv

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#327]
**Description**

Currently, static frame can write an empty frame with named index to tsv, but cannot read the same frame from tsv. It would be nice if it could. 

**Example**

Provide code or pseudo code demonstrating how the feature might work, or how current behavior might be changed.

```python
import static_frame as sf
import io

f = sf.Frame(columns=['column'], index=sf.Index([], name='index'))
s = io.StringIO()
f.to_tsv(s)
s.seek(0)

sf.Frame.from_tsv(s, index_depth=1)
# static_frame.core.exception.ErrorInitFrame: no data from which to extract index_depth 1
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
import static_frame as sf
sf.Platform.display()

<Series: platform>
<Index>
platform           Linux-5.4.0-66-generic-x86_64-with
sys.version        3.8.3 (default, Mar  8 2021, 09:23:44) [GCC 7.5.0]
static-frame       0.8.4
numpy              1.17.4
pandas             0.24.2
xlsxwriter         1.1.2
openpyxl           2.5.4
xarray             <ModuleNotFoundError>
tables             3.6.1
pyarrow            0.17.0
msgpack            <ModuleNotFoundError>
msgpack_numpy      <ModuleNotFoundError>
<<U13>             <object>
```

**Comment by ForeverWintr** [comment]
Commenting as I think this may be related: I think `Frame.from_tsv` doesn't read index name at all, currently. It would be nice if it did!

```python
f = sf.Frame.from_element('asdf', index=sf.Index([1,2,3], name='Name'), columns=['a'])
s = io.StringIO()
f.to_tsv(s)
s.seek(0)
f2 = sf.Frame.from_tsv(s, index_depth=1)
f2.index.name #None
```

**Comment by flexatone** [comment]
Thanks for isolating this issue. This is fixed and will be part of the 0.8.7 release. Note that reading in the name of index or columns is not enabled by default, but is available with the `index_name_depth_level` and `columns_name_depth_level` parameters, respectively, as shown below.

```python
>>> f1 = sf.Frame(columns=['column'], index=sf.Index([], name='index'))                                                                
>>> s = io.StringIO()                                                                                                                  
>>> f1.to_tsv(s)                                                                                                                       
                                                                                                                                                                                                                                             
>>> s.seek(0)                                                                                                                          
0
>>> sf.Frame.from_tsv(s, index_depth=1, index_name_depth_level=0)                                                                      
<Frame>
<Index>        column    <<U6>
<Index: index>
<int64>        <float64>

>>> s.seek(0)                                                                                                                          
0
>>> sf.Frame.from_tsv(s, index_depth=1, columns_name_depth_level=0)                                                                    
<Frame>
<Index: index> column    <<U6>
<Index>
<int64>        <float64>
```
